packages/components/src/components/carousel/carousel.tsx

```diff
@@ -309,13 +323,17 @@ export class Carousel {
 
   private registerSplideHandlers(splide: Splide): void {
     splide.on('mounted', () => {
-      updatePrevNextButtons(this.btnPrev, this.btnNext, splide);
-      renderPagination(this.paginationEl, this.amountOfPages, this.activeSlideIndex); // initial pagination
+      if (this.isDraggable) {
```

Comment:
Might be safer to read this from `splide.options` to avoid or uncover a sync mismatch where the component is in a different state than the splide instance.